lib/rest-router.js

```diff
@@ -0,0 +1,66 @@
+
+var url = require('url');
+var http = require('http');
+var Trie = require('./trie');
+
+var methods = getCurrentNodeMethods();
```

Comment:
since `methods` should be treated as a constant, please use all-capital letters in the name. I would add `methods` to the name, to make the distinction between HTTP methods and shared/remote methods clear.

```js
var HTTP_METHODS = getHttpMethods();
```